test/integration/ovs/ofctrl_test.go

```diff
@@ -712,6 +713,113 @@ func TestMoveTunMetadata(t *testing.T) {
+	defer DeleteOVSBridge(br)
+
+	bridge := newOFBridge(br)
+	table = bridge.CreateTable(2, 3, binding.TableMissActionNext)
+
+	err = bridge.Connect(maxRetry, make(chan struct{}))
+	require.Nil(t, err, "Failed to start OFService")
+	defer bridge.Disconnect()
+
+	ofctlClient := ovsctl.NewClient(br)
+	ctIpSrc := net.ParseIP("1.1.1.1")
+	ctIpDst := net.ParseIP("2.2.2.2")
+	_, ctIpSrcNet, _ := net.ParseCIDR("1.1.1.0/24")
+	_, ctIpDstNet, _ := net.ParseCIDR("2.2.2.0/24")
+	ctPortSrc := uint16(10001)
+	ctPortDst := uint16(20002)
+	priority := uint16(200)
+	flow1 := table.BuildFlow(priority).
+		MatchProtocol(binding.ProtocolIP).
+		MatchCTStateNew(true).
+		MatchCTSrcIP(ctIpSrc).
+		MatchCTDstIP(ctIpDst).
+		MatchCTSrcPort(ctPortSrc).
+		MatchCTDstPort(ctPortDst).
+		MatchCTProtocol(binding.ProtocolTCP).
+		Action().ResubmitToTable(table.GetNext()).
+		Done()
+	flow2 := table.BuildFlow(priority).
+		MatchProtocol(binding.ProtocolIP).
+		MatchCTStateEst(true).
+		MatchCTSrcIPNet(*ctIpSrcNet).
+		MatchCTDstIPNet(*ctIpDstNet).
+		MatchCTProtocol(binding.ProtocolTCP).
+		Action().ResubmitToTable(table.GetNext()).
+		Done()
+	expectFlows := []*ExpectFlow{
+		{fmt.Sprintf("priority=%d,ct_state=+new,ct_nw_src=%s,ct_nw_dst=%s,ct_nw_proto=6,ct_tp_src=%d,ct_tp_dst=%d,ip",
+			priority, ctIpSrc.String(), ctIpDst.String(), ctPortSrc, ctPortDst),
+			fmt.Sprintf("resubmit(,%d)", table.GetNext()),
+		},
+		{
+			fmt.Sprintf("priority=%d,ct_state=+est,ct_nw_src=%s,ct_nw_dst=%s,ct_nw_proto=6,ip",
+				priority, ctIpSrcNet.String(), ctIpDstNet.String()),
+			fmt.Sprintf("resubmit(,%d)", table.GetNext()),
+		},
+	}
+	for _, f := range []binding.Flow{flow1, flow2} {
+		err = f.Add()
+		assert.Nil(t, err, "no error returned when adding flow")
+	}
+	CheckFlowExists(t, ofctlClient, uint8(table.GetID()), true, expectFlows)
+	for _, f := range []binding.Flow{flow1, flow2} {
+		err = f.Delete()
+		assert.Nil(t, err, "no error returned when deleting flow")
+	}
+	CheckFlowExists(t, ofctlClient, uint8(table.GetID()), false, expectFlows)
+}
+
+func TestNoteAction(t *testing.T) {
+	br := "br10"
+	err := PrepareOVSBridge(br)
+	require.Nil(t, err, fmt.Sprintf("Failed to prepare OVS bridge: %v", err))
+	defer DeleteOVSBridge(br)
+
+	bridge := newOFBridge(br)
+	table = bridge.CreateTable(2, 3, binding.TableMissActionNext)
+
+	err = bridge.Connect(maxRetry, make(chan struct{}))
+	require.Nil(t, err, "Failed to start OFService")
+	defer bridge.Disconnect()
+
+	ofctlClient := ovsctl.NewClient(br)
+	priority := uint16(1001)
+	srcIP := net.ParseIP("1.1.1.2")
+	testNotes := "test for noteActions."
+	byteSlice := make([]byte, (len(testNotes)/8)*8+6)
+	copy(byteSlice, testNotes)
+	var bytesStrs []string
+	for i := range byteSlice {
+		if byteSlice[i] < 16 {
+			bytesStrs = append(bytesStrs, fmt.Sprintf("0%x", byteSlice[i]))
+		} else {
+			bytesStrs = append(bytesStrs, fmt.Sprintf("%x", byteSlice[i]))
+		}
+	}
+	notesStr := strings.Join(bytesStrs, ".")
```

Comment:
could you group this into a local function to make it clear what you are doing:
```
convertNoteToHex := func(note string) string {
    // ...
    return bytesStr
}
```

Reply:
Done.